src/Filters/WhereNull.php

```diff
@@ -0,0 +1,100 @@
+<?php
+/*
+ * Copyright 2022 Cloud Creativity Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+declare(strict_types=1);
+
+namespace LaravelJsonApi\Eloquent\Filters;
+
+use LaravelJsonApi\Core\Support\Str;
+use LaravelJsonApi\Eloquent\Contracts\Filter;
+
+class WhereNull implements Filter
+{
+    use Concerns\HasColumn;
+    use Concerns\IsSingular;
+
+    /**
+     * @var bool
+     */
+    private bool $not = false;
+
+    /**
+     * @var string
+     */
+    private string $key;
+
+    /**
+     * Create a new filter.
+     *
+     * @param string      $key
+     * @param string|null $column
+     *
+     * @return static
+     */
+    public static function make(string $key, string $column = null): self
+    {
+        return new static($key, $column);
+    }
+
+    /**
+     * WhereNotNull constructor.
+     *
+     * @param string|null $column
+     */
+    public function __construct(string $key, string $column = null)
+    {
+        $this->key = $key;
+        $this->column = $column ?: $this->guessColumn();
+    }
+
+    /**
+     * @return string
+     */
+    public function key(): string
+    {
+        return $this->key;
+    }
+
+    /**
+     * @return $this
+     */
+    public function not(): WhereNull
+    {
+        $this->not = true;
+        return $this;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function apply($query, $value)
+    {
+        if ($value == is_bool($value)) {
+            $this->not = (bool) $value;
```

Comment:
Sorry I didn't pick up on this when I commented earlier, but I now understand what you're trying to do here. Yes, we definitely want the client to specify whether it wants to find null values or not null values.

Remove the `$not` property from this class, it is not required.

Then here, in the `apply` method, use the following:

```php
$not = filter_var($value, FILTER_VALIDATE_BOOL);
```

then use that variable when calling `$query->whereNull()`. We need to use `filter_var` because the value provided by the client comes through as a string in query parameters.

it would be useful to know what your use of this filter is in real-life, so I can check this matches how you'd use it? The change I'm proposing would mean if a client provided `true` then null values would be returned, if they provided `false` then non-null values would be provided.

Reply:
I have to note that when using FILTER_VALIDATE_BOOL, the php level requirement will need to be updated to php8 in order to use this.

Reply:
Yeah that filter works in the current version (it is used in other places) - there's a polyfill for it. Thanks for making the changes, I'll hopefully get this merged soon,

Reply:
awesome!